inference-engine/tests/functional/plugin/cpu/single_layer_tests/lstm_cell.cpp

```diff
@@ -0,0 +1,129 @@
+        LayerTestsDefinitions::LSTMCellParams basicParamsSet;
+        std::map<std::string, std::string> enforceBF16;
+
+        std::tie(basicParamsSet, cpuParams, enforceBF16) = obj.param;
+        std::ostringstream result;
+
+        result << LayerTestsDefinitions::LSTMCellTest::getTestCaseName(testing::TestParamInfo<LayerTestsDefinitions::LSTMCellParams>(
+                basicParamsSet, 0));
+        result << CPUTestsBase::getTestCaseName(cpuParams);
+
+        if (!enforceBF16.empty()) {
+            result << "_PluginConf";
+            for (auto& item : enforceBF16) {
+                if (item.second == PluginConfigParams::YES)
+                    result << "_" << item.first << "=" << item.second;
+            }
+        }
+        return result.str();
+    }
+
+protected:
+    void SetUp() {
+        LayerTestsDefinitions::LSTMCellParams basicParamsSet;
+        CPUSpecificParams cpuParams;
+        std::map<std::string, std::string> enforceBF16;
+
+        bool should_decompose;
+        size_t batch;
+        size_t hidden_size;
+        size_t input_size;
+        std::vector<std::string> activations;
+        std::vector<float> activations_alpha;
+        std::vector<float> activations_beta;
+        float clip;
+        InferenceEngine::Precision netPrecision;
+        threshold = 0.05;
+
+        std::tie(basicParamsSet, cpuParams, enforceBF16) = this->GetParam();
+        std::tie(inFmts, outFmts, priority, selectedType) = cpuParams;
+        std::tie(should_decompose, batch, hidden_size, input_size, activations, clip, netPrecision, targetDevice) = basicParamsSet;
+
+        configuration.insert(enforceBF16.begin(), enforceBF16.end());
+
+        std::vector<std::vector<size_t>> inputShapes = {
+            {{batch, input_size}, {batch, hidden_size}, {batch, hidden_size}, {4 * hidden_size, input_size}, {4 * hidden_size, hidden_size}, {4 * hidden_size}},
+        };
+        auto ngPrc = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(netPrecision);
+        auto params = ngraph::builder::makeParams(ngPrc, {inputShapes[0], inputShapes[1], inputShapes[2]});
+        std::vector<ngraph::Shape> WRB = {inputShapes[3], inputShapes[4], inputShapes[5]};
+        auto lstm_cell = ngraph::builder::makeLSTM(
+            ngraph::helpers::convert2OutputVector(ngraph::helpers::castOps2Nodes(params)), WRB, hidden_size, activations, {}, {}, clip);
+        ngraph::ResultVector results{std::make_shared<ngraph::opset1::Result>(lstm_cell->output(0)),
+                                     std::make_shared<ngraph::opset1::Result>(lstm_cell->output(1))};
+        function = std::make_shared<ngraph::Function>(results, params, "lstm_cell");
+        if (should_decompose) {
+            ngraph::pass::Manager m;
+            m.register_pass<ngraph::pass::LSTMCellDecomposition>();
+            m.run_passes(function);
+        }
+
+        if (selectedType.empty()) {
+            selectedType = getPrimitiveType();
+        }
```

Comment:
It is useless code since you always sets selectedType through  CPUSpecificParams